crates/bifrost/src/providers/replicated_loglet/loglet.rs

```diff
@@ -260,6 +260,15 @@ impl<T: TransportConnect> Loglet for ReplicatedLoglet<T> {
         )
         .run(self.networking.clone())
         .await?;
+        // If we are the sequencer, we need to wait until the sequencer is drained.
+        if let SequencerAccess::Local { handle } = &self.sequencer {
+            handle.drain().await?;
+            self.known_global_tail.notify_seal();
+        };
+        // On remote sequencer, we only set our global tail to sealed when we call find_tail and it
+        // returns Sealed. We should NOT:
+        // - Use AppendError::Sealed to mark our sealed global_tail
+        // - Mark our global tail as sealed on successful seal() call.
```

Comment:
Why would these two cases be problematic? Is it because the sequencer returns an `AppendError::Sealed` before the loglet is fully sealed (e.g. 1 log server currently being sealed and the sequencer is hitting this one)?

Reply:
Yes. Rejecting appends with Sealed doesn't automatically mean that the current value of known_global_tail is safe to be considered sealed.

Reply:
Wouldn't the message carrying the error from the sequencer back to the callser also contain the sealed offset?

Reply:
Note that the sequencer might send AppendError::Sealed during drain, that doesn't mean that it's knowledge of global tail should be used as the reliable "seal" tail.